src/migration/1610566573713-MigrateTables.ts

```diff
@@ -0,0 +1,96 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class MigrateTables1610566573713 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+
+        // Create a function to drop a pre-existing index whose name we do not know
+        await queryRunner.query(
+            `CREATE OR REPLACE FUNCTION drop_index(_tablename TEXT, _colname TEXT)
+              RETURNS INTEGER
+            AS $$
+            DECLARE
+              i RECORD;
+            BEGIN
+              FOR i IN
+                (SELECT indexname FROM pg_indexes WHERE tablename = _tablename AND LOWER(indexname) LIKE 'idx%' AND LOWER(indexdef) LIKE CONCAT('%', LOWER(_colname), '%'))
+              LOOP
+                RAISE INFO 'DROPPING INDEX: %', i.indexname;
+                EXECUTE 'DROP INDEX IF EXISTS ' || i.indexname;
+              END LOOP;
+            RETURN 1;
+            END;
+            $$ LANGUAGE plpgsql;`
+        );
+
+        // Remove unnecessary old indexes from the synchronized version of the database
+        await queryRunner.query(
+            `SELECT drop_index('sms_otp', 'agent_id');`
+        );
+        await queryRunner.query(
+            `SELECT drop_index('wallet_credentials', 'did');`
+        );
+
+        // Don't need this function anymore, so delete it
+        await queryRunner.query(
+            `DROP FUNCTION drop_index;`
+        );
+
+        // Create a function to rename a unique or primary key index whose name we do not know
+        await queryRunner.query(
+            `CREATE OR REPLACE FUNCTION rename_constraint(_tablename TEXT, _colname TEXT, _newname TEXT, _pk BOOLEAN)
+              RETURNS INTEGER
+            AS $$
+            DECLARE
+              i RECORD;
+            BEGIN
+              IF _pk THEN
+                  FOR i IN
+                    (SELECT indexname FROM pg_indexes WHERE tablename = _tablename AND LOWER(indexname) LIKE 'pk%' AND LOWER(indexdef) LIKE CONCAT('%', LOWER(_colname), '%'))
+                  LOOP
+                    RAISE INFO 'RENAMING CONSTRAINT % to %', i.indexname, _newname;
+                    EXECUTE 'ALTER TABLE IF EXISTS ' || _tablename || ' RENAME CONSTRAINT ' || i.indexname || ' TO ' || _newname;
+                  END LOOP;
+              ELSE
+                  FOR i IN
+                    (SELECT indexname FROM pg_indexes WHERE tablename = _tablename AND LOWER(indexname) LIKE 'uq%' AND LOWER(indexdef) LIKE CONCAT('%', LOWER(_colname), '%'))
+                  LOOP
+                    RAISE INFO 'RENAMING INDEX % to %', i.indexname, _newname;
+                    EXECUTE 'ALTER TABLE IF EXISTS ' || _tablename || ' RENAME CONSTRAINT ' || i.indexname || ' TO ' || _newname;
+                  END LOOP;
+              END IF;
+            RETURN 1;
+            END;
+            $$ LANGUAGE plpgsql;`
+        );
+
+        // Give old PKs and Unique Keys a predictable name
```

Comment:
👍 Nice, this is a good way of solving the issue Esmaiel saw before. Note that you may need to do some custom dropping and adding of indexes on our deployed DBs if you want to use these same names there.

Reply:
I looked into the indexes in dev and wrote these functions based on the names I saw there. I'm assuming the tables in the deployed DBs also were created via typeorm's sycronization, in which case they'd have similarly named indexes and this migration should work just fine. But it would be great to verify in those environments! I just need the right credentials. :)